devito/finite_differences/derivative.py

```diff
@@ -269,7 +269,8 @@ def _eval_at(self, func):
         setup where one could have Eq(u(x + h_x/2), v(x).dx)) in which case v(x).dx
         has to be computed at x=x + h_x/2.
         """
-        x0 = dict(func.indices_ref._getters)
+        # If an x0 already exists do not overwrite it
+        x0 = self.x0 or dict(func.indices_ref._getters)
```

Comment:
btw, remind me what's the difference between func.indices._getters and func.dimensions ?

Reply:
It's a dict `{dim: index_ref}`. So if not staggered is `{d: d fr d in dimensions}` and for staggered `{d: d+ .5spacing if stagg is d else x fir d in dimensions}`